File: src/701-799/704. Binary Search/Solution.js
/**
 * @param { number[] } nums
 * @param { number } target
 * @return { number }
 */
const search = function (nums, target) {
	let left = 0
	let right = nums.length - 1

	while (left <= right) {
		let mid = Math.round((left + right) / 2)

		if (nums[mid] === target) return mid

		if (nums[mid] > target) {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}

	return -1
}

File: src/701-799/704. Binary Search/Solution.py
from typing import List


class Solution:
    @staticmethod
    def search(nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1

        while left <= right:
            mid = (left + right) // 2

            if nums[mid] == target:
                return mid

            if nums[mid] > target:
                right = mid - 1
            elif nums[mid] < target:
                left = mid + 1

        return -1
